public/service/message.go

```diff
@@ -453,6 +482,41 @@ func (b MessageBatch) BloblangQuery(index int, blobl *bloblang.Executor) (*Messa
 	return nil, nil
 }
 
+// BloblangQueryValue executes a parsed Bloblang mapping on a message batch,
+// from the perspective of a particular message index, and returns the result or
+// an error if the mapping fails.
+//
+// This method allows mappings to perform windowed aggregations across message
+// batches.
+func (b MessageBatch) BloblangQueryValue(index int, blobl *bloblang.Executor) (any, error) {
+	uw := blobl.XUnwrapper().(interface {
+		Unwrap() *mapping.Executor
+	}).Unwrap()
+
+	msg := make(message.Batch, len(b))
+	for i, m := range b {
+		msg[i] = m.part
+	}
```

Comment:
Is this necessary here? I left it in since I don't know exactly why `BloblangQuery()` does it, but I'm guessing in that case it's relevant because it returns messages.

Reply:
Yeah we need to provide the batch as execution context, it's how the `.from` methods work.